pkg/apis/scorecard/v1alpha2/types.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// State is a type used to indicate the result state of a Test.
+type State string
+
+const (
+	// UnsetState is the default state for a TestResult. It must be updated by UpdateState or by the Test.
+	UnsetState State = "unset"
+	// PassState occurs when a Test's ExpectedPoints == MaximumPoints.
+	PassState State = "pass"
+	// FailState occurs when a Test's ExpectedPoints == 0.
+	FailState State = "fail"
+	// ErrorState occurs when a Test encounters a fatal error and the reported points should not be considered.
+	ErrorState State = "error"
+)
+
+// ScorecardSuiteResult contains the combined results of a suite of tests.
+// +k8s:openapi-gen=true
+type ScorecardSuiteResult struct {
```

Comment:
To expand on this, I feel like `v1alpha2` types are probably more complex than is really necessary.

At the end of the day, what I think we want to output is:
1. A list of test results
2. The logs from the overall run, if failure or verbose or JSON (which I think is similar to how `go test` behaves).

I think for each test result we would output:
1. Name
2. State (pass/fail/error)
3. Description, if verbose or JSON
4. Labels, if verbose or JSON (I know we're not to the label work yet, but it might be good to go ahead and add this to the type)
5. Errors, if any
6. Suggestions, if any

Reply:
We would also need to capture the overall outcome of the scorecard run. Is it overall failed vs. pass etc.

Reply:
I was going to hold off on doing any label work in this PR since its pretty large already, I've added the overall outcome to the v1alpha2 api based on this comment.

Reply:
@jmccormick2001 Since we're making `v1alpha2` the default as of this PR, I think we should try to get it's type definition right, so we don't have to go back and change it when we get to the rest of the labels features. I'd suggest doing three things in this PR:
1. Add a labels field to `v1alpha2.ScorecardTestResult` as a placeholder (we won't populate them yet, however)
2. Remove the suite layer from `v1alpha2` entirely
3. Make sure the JSON output format spec contains everything we'll need to finish the epic. If we aren't confident we can do this piece yet, I would hesitate to make `v1alpha2` the default.

@dmesser  What do you think is sufficient to capture the overall output of the scorecard run? Do you think the log and a simple `pass`/`fail`/`error` state field? Anything else?

Basically, would we be alright with a top-level output struct that contains?
```go
type ScorecardOutput struct {
    Results []ScorecardTestResult
    State string // pass, fail, or error
    Log string
}

type ScorecardTestResult struct {
    Name string
    State string
    Description string
    Labels map[string]string
    Errors []error
    Suggestions []string
}
```
WDYT?

Reply:
sounds good, I was going to add the labels field in the upcoming label story work but could do that in this PR if you want.  I think the notion of the 'suite' was not clear in my mind how it would be implemented or if it totally removed as a concept, but rather it just becomes another label in the map that is populated.